test/MusicStore.Test/AnnouncementComponentTest.cs

```diff
@@ -0,0 +1,117 @@
+
+        public AnnouncementComponentTest()
+        {
+            var services = new ServiceCollection();
+
+            services.AddEntityFramework()
+                      .AddInMemoryStore()
+                      .AddDbContext<MusicStoreContext>();
+
+            _serviceProvider = services.BuildServiceProvider();
+        }
+
+        [Fact]
+        public async Task AnnouncementComponent_Returns_LatestAlbum()
+        {
+            // Arrange
+            var today = DateTime.Parse("10/30/2012");
+
+            var announcementComponent = new AnnouncementComponent()
+            {
+                DbContext = _serviceProvider.GetRequiredService<MusicStoreContext>(),
+                Cache = _serviceProvider.GetRequiredService<IMemoryCache>(),
+                Clock = new TestSystemClock(today),
+            };
+
+            PopulateData(announcementComponent.DbContext, latestAlbumDate: today);
+
+            // Action
+            var result = await announcementComponent.InvokeAsync();
+
+            // Assert
+            Assert.NotNull(result);
+            var viewResult = Assert.IsType<ViewViewComponentResult>(result);
+            Assert.Null(viewResult.ViewName);
+            var albumResult = Assert.IsType<Album>(viewResult.ViewData.Model);
+            Assert.Equal(today, albumResult.Created.Date);
+        }
+
+        private static void PopulateData(DbContext context, DateTime latestAlbumDate)
+        {
+            var albums = TestAlbumDataProvider.GetAlbums(latestAlbumDate);
+
+            foreach (var album in albums)
+            {
+                context.Add(album);
+            }
+
+            context.SaveChanges();
+        }
+
+        private class TestAlbumDataProvider
+        {
+            public static Album[] GetAlbums(DateTime latestAlbumDate)
+            {
+                var generes = Enumerable.Range(1, 10).Select(n =>
+                    new Genre()
+                    {
+                        GenreId = n,
+                        Name = "Genre Name " + n,
+                    }).ToArray();
+
+                var artists = Enumerable.Range(1, 10).Select(n =>
+                    new Artist()
+                    {
+                        ArtistId = n + 1,
+                        Name = "Artist Name " + n,
+                    }).ToArray();
+
+                var albums = Enumerable.Range(1, 10).Select(n =>
+                    new Album()
+                    {
+                        Artist = artists[n - 1],
+                        ArtistId = n,
+                        Genre = generes[n - 1],
+                        GenreId = n,
+                        Created = latestAlbumDate.AddDays(1 - n),
+                    }).ToArray();
+
+                return albums;
+            }
+        }
+
+        private class TestSystemClock : ISystemClock
+        {
+            private DateTime _date;
+
+            public TestSystemClock(DateTime date)
```

Comment:
This is because the type of Album's Create property is DateTime, but ISystemClock interace returns DateTimeOffset
https://github.com/aspnet/MusicStore/blob/master/src/MusicStore/Models/Album.cs#L36-L38
So, they shoud be converted at some point either here or when populating test data. 
Generally we should use DateTimeOffset over DateTime, but I think it is fine with the Album model since we are concerned about Date for this model. (https://msdn.microsoft.com/en-us/library/bb384267(v=vs.110).aspx)

Let me know if you want to convert it when populating test data instead of in this interface.